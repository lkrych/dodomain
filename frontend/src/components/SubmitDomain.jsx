import React, {Component} from 'react';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  root: theme.mixins.gutters({
    paddingTop: theme.spacing.unit * 3,
    paddingBottom: 16,
  }),
});

class SubmitDomain extends Component {
  constructor(props){
    super(props);
    this.state = {
     
      name: '',
      description: ''
    
    };
    this.onSubmit = this.onSubmit.bind(this);
  }

  onInput (property) {
    return (e) => {
      this.setState({ [property]: e.target.value });
    };
  }

  onSubmit(e){
    e.preventDefault();
    this.props.submitDomain({domain: this.state});
    this.setState({
      name: '',
      description: ''
    });
  }

  render(){
    const { classes } = this.props;
    let textErrors = <div className="errors">{this.props.errors}</div>;

    return (
      <Paper className={classes.root} elevation={4}>
      <Typography variant="headline" gutterBottom>
          Submit a Domain
        </Typography>
        <Typography variant="body2" gutterBottom>
          Domain Name ex: "npr.org"
        </Typography> 
        {textErrors}
        <form action="">
          <div className='field'>
            <TextField
              label='Domain Name' 
              type='name'
              id='name' 
              onChange={this.onInput('name')}
              value={this.state.name} />
          </div>

        <div className='field'>
          <TextField
            id="description"
            value={this.state.description}
            onChange={this.onInput('description')} 
            name='description' 
            label="Description of domain"
            placeholder="Description of domain"
            multiline
            className={classes.textField}
            margin="normal"
          />
         
        </div>


        <Button
          type="submit"
          variant="contained" color="primary"
          className="btn btn-primary btn-submit"
          onClick={this.onSubmit}>
          Submit Domain
        </Button>
        </form>
      </Paper>
    );
  }
}

export default withStyles(styles)(SubmitDomain);