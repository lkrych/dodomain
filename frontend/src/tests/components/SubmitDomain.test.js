import React from 'react';
import { shallow } from 'enzyme';

import SubmitDomain from '../../components/SubmitDomain';

describe('SubmitDomain', () => {
  const mockSubmitForm = jest.fn();
  const submitDomain = shallow(<SubmitDomain submitDomain={mockSubmitForm} />);
  
  it('renders correctly', () => {
    expect(submitDomain).toMatchSnapshot();
  });

  it('correctly initializes the `state` with an empty name and description', () => {
    expect(submitDomain.state().name).toEqual('');
    expect(submitDomain.state().description).toEqual('');
  });

  describe('when typing into the name input', () => {
    const name = 'Noodle@gmail.com';

    beforeEach(() => {
      submitDomain.find('#name').simulate('change', { target: { value: name} });
    });

    it('updates the person in `state`', () => {
      expect(submitDomain.state().name).toEqual(name);
    });
  });

  describe('when typing into the description input', () => {
    const description = 'description';

    beforeEach(() => {
      submitDomain.find('#description').simulate('change', { target: { value: description} });
    });

    it('updates the person in `state`', () => {
      expect(submitDomain.state().description).toEqual(description);
    });
  });

  describe('on submission of form', () => {
    const mockedEvent = { target: {}, preventDefault: jest.fn() };

    it('should call the login method', () => {
      beforeEach(() => {
        submitDomain.find('.btn-submit').simulate('click', mockedEvent);
      });

      it('calls the login callback', () => {
        expect(mockSubmitForm.mock.calls).toEqual(1);
      });
    });
  });

  describe('Error Text', () => {
    it('should be an empty string if there are no session_errors', () => {
      expect(submitDomain.find('.errors').html()).toEqual("<div class=\"errors\"></div>");
    });
    
    let submitDomainWithErrors = shallow(<SubmitDomain submitDomain={mockSubmitForm} errors={"there is an error"}/>);
    it('should be populated if there are session_errors', () => {
      expect(submitDomainWithErrors.find('.errors').html()).toEqual("<div class=\"errors\">there is an error</div>");
    });
  });
});