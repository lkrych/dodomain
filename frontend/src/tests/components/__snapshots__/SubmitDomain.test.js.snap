// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitDomain renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(SubmitDomain)
    submitDomain={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "root": "SubmitDomain-root-1",
      },
      "submitDomain": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "root": "SubmitDomain-root-1",
        },
        "submitDomain": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
